events/api.py

```diff
@@ -1938,6 +1938,18 @@ def _filter_event_queryset(queryset, params, srs=None):
 
         queryset = queryset.filter(qset)
 
+    val = params.get('event_type', None)
+    if val:
+        vals = val.lower().split(',')
+        event_types = {k[1].lower(): k[0] for k in Event.TYPE_IDS}
+        search_vals = []
+        for v in vals:
+            if v not in event_types:
+                raise ParseError(_(f'Event type can be of the following values:{" ".join(event_types.keys())}'))
+            search_vals.append(event_types[v])
+
+        queryset = queryset.filter(type_id__in=search_vals)
```

Comment:
I would create a function of this - 1 function per parameter-feature -handling. If the handlers are repetitive, then make it generic enough to serve all the kind.

Reply:
that's the style of the whole file, then I would need to refactor pretty much all of it.